### ISSUE TITLE
Getting Unknown Exception when Deploying or Retrieving

### ISSUE THREAD

**Issue by mani-radhu** [forcedotcom/salesforcedx-vscode#3137]
"14:56:08.399 Starting SFDX: Retrieve Source from Org

Unknown Exception"

My system is running behind Https & http Proxy, Also I am tring to Retrieve Resources from Metadata.
Cant find any solution online
My sfdx version is as below :

PS C:\Users\m347654\Documents\GIT\Protein Avenger VSCode\FEATURE\feature-testftrox> sfdx plugins --core
@oclif/plugin-autocomplete 0.3.0 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.2 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.10.0 (core)
@oclif/plugin-update 1.4.0-3 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-trust 3.6.0 (core)
alias 1.1.8 (core)
auth 1.5.1 (core)
config 1.2.7 (core)
generator 1.1.5 (core)
salesforcedx 51.7.0 (core)
├─ schema 1.0.4 (core)
├─ data 0.4.3 (core)
├─ apex 0.1.20 (core)
├─ org 1.6.1 (core)
├─ templates 51.3.0 (core)
├─ salesforce-alm 51.6.8 (core)
├─ user 1.2.7 (core)
├─ custom-metadata 1.0.11 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
└─ limits 1.0.4 (core)
sfdx-cli 7.96.0 (core)
telemetry 1.1.1 (core)

Also i did set force:config: tp Api version 51 as i had faced similiar issue earlier
PS C:\Users\m347654\Documents\GIT\Protein Avenger VSCode\FEATURE\feature-testftrox> sfdx force:config:get apiVersion
WARNING: apiVersion configuration overridden at "51.0"
=== Get Config
Name Value Success
────────── ───── ───────
apiVersion 51.0 true

Please check the reference ticket raised with vscode 
https://github.com/microsoft/vscode/issues/120867

![image](https://user-images.githubusercontent.com/82180969/114127462-0b8f1000-98c0-11eb-8b6c-87f07543c10d.png)

Please help me out on this.

Thanks
Mani

**Comment by sfsholden** [comment]
@mani-radhu Thank you for reaching out to us.

Is there any additional information listed in the below resource?
Click `Help` from the top menu pane in VSCode -> Select `Toggle Developer Tools` -> Select `Console`


Please also list the following:
- VS Code version: 
- OS version: 
- SFDX CLI version (`sfdx -v`): 


Potential Workarounds:

1. If you disable the setting `Experimental: Deploy Retrieve` does this unblock you?
https://developer.salesforce.com/tools/vscode/en/user-guide/perf-enhancements/#setup

2. We did also release version 51.8.0 of the extensions yesterday. Please try updating to this version to see if your issue is resolved.

**Comment by mani-radhu** [comment]
Awesome @sfsholden I am able to work as usual .

Thanks a ton 

I followed step 1.

**Comment by sfsholden** [comment]
@mani-radhu Glad to hear you are unblocked!

Was there any additional information from the Developers Tools console? It looks like we might still have an issue to resolve around proxies for the performance improvement route. Any further info would help us in determining what is causing this. 

Also just for a sanity check - did you happen to still see the same issue for extensions 51.8.0 with the setting enabled?


Thanks!

**Comment by mani-radhu** [comment]
I will cross check and confirm you

**Comment by brpowell** [comment]
Hi @mani-radhu - I'm also curious what metadata types you were trying to retrieve if you happen to remember. Thanks!

**Comment by mani-radhu** [comment]
Kindly find my findings after enabling "Experimental: Deploy Retrieve"

While loading VSCode its throwing an error
![image](https://user-images.githubusercontent.com/82180969/114430339-f34f1780-9b83-11eb-876a-85e3f97950a0.png)

Kindly find the below issue while loading 

![image](https://user-images.githubusercontent.com/82180969/114437566-55ac1600-9b8c-11eb-8ceb-35ac0b0e8a71.png)

Its working fine after disabling "Experimental: Deploy Retrieve", no errors while loading Vscode as well

Thanks again for the help on time cheers !!!!!

**Comment by mani-radhu** [comment]
Regarding Metadata, as of now the coding components, ApexClass, ApexTrigger, AuraDefinitionBundle,LightningComponentBundle

**Comment by smaddox-sf** [comment]
See #3152 for report of another similar issue with 'unknown exception' when attempting to deploy or retrieve using a manifest.  (3152 closed as a duplicate)

**Comment by brpowell** [comment]
Closing this since the latest release no longer reports "unknown exception" when an issue has been encountered. Instead it should surface a more descriptive error. Many of these errors may simply be malformed file names for metadata ("could not infer a metadata type"). If it's another error that doesn't have a clear path to be unblocked, feel free to open a new issue so we can address that particular error.
